bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/FreeDeviceIntf.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeboxos.internal.handler;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+
+/**
+ * The {@link FreeDeviceIntf} defines some common methods for various devices (server, player, repeater) not belonging
+ * to the same class hierarchy
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public interface FreeDeviceIntf extends ApiConsumerIntf {
+    public ChannelUID getEventChannelUID();
+
+    public void triggerChannel(ChannelUID channelUID, String event);
+
+    default long checkUptimeAndFirmware(long newUptime, long oldUptime, String firmwareVersion) {
+        if (newUptime < oldUptime) {
+            triggerChannel(getEventChannelUID(), "restarted");
+            Map<String, String> properties = editProperties();
+            if (!firmwareVersion.equals(properties.get(Thing.PROPERTY_FIRMWARE_VERSION))) {
+                properties.put(Thing.PROPERTY_FIRMWARE_VERSION, firmwareVersion);
+                updateProperties(properties);
+                triggerChannel(getEventChannelUID(), "firmware_updated");
+            }
+        }
+        return newUptime;
+    }
+
+    default void processReboot(Runnable actualReboot) {
+        triggerChannel(getEventChannelUID(), "reboot_requested");
+        actualReboot.run();
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DUTY_CYCLE, "System rebooting...");
+        stopRefreshJob();
+        getScheduler().schedule(this::initialize, 30, TimeUnit.SECONDS);
```

Comment:
This task needs to be cancelled in `dispose()`.

Reply:
Correct, I reviewed the way I handle scheduledJobs globally.